### DISCUSSION TITLE
Access to font at has been blocked by CORS policy.

### DISCUSSION THREAD

**Discussion by Maxet24** [vercel/next.js#32069] [Help] [answered]
### What version of Next.js are you using?

12.0.4

### What version of Node.js are you using?

Vercel Current (sorry, didn't find)

### What browser are you using?

Chrome

### What operating system are you using?

Windows

### How are you deploying your application?

Vercel

### Describe the Bug

Hello! I have two fonts that don't work on my site. In the google chrome console I get an error like this:
```
Access to font at 'webpack:///mini-css-extract-plugin/_next/static/media/Gardene-Stone.b0f685ba.ttf' from
origin 'https://somesite.club' has been blocked by CORS policy: 
Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https.
```
<img width="640" alt="Снимок экрана 2021-11-27 135708" src="https://user-images.githubusercontent.com/62026101/143678445-563669bb-39c2-4db6-9f1f-b5df348963f6.png">
If I run the site using `"npm run dev"`, everything works fine, but if I upload it to the hosting, the font stays standard and the JS console gives an error.

This is the css code that drives the font.
```
@font-face {
    font-family: 'GardeneStone';
    src: url('../fonts/GardeneStone.ttf') format('truetype'),
        url('../fonts/GardeneStone.woff') format('woff'); /* Modern Browsers */
    font-weight: 300;
    font-style: regular;
    font-display: swap;
}

    .GardeneStone {
    font-family: GardeneStone, sans-serif;
}
```

### Expected Behavior

Two working fonts.

### To Reproduce

[GitHub Rep](https://github.com/Maxet24/crazy-suit-cases)

**Comment by balazsorban44** [comment]
Your repository is private.

**Comment by Maxet24** [comment]
> Your repository is private.

@balazsorban44 
Oh, sorry, i changed.

**Comment by Maxet24** [comment]
> Your repository is private.

So what do you think?

**Comment by mattcarlotta** [comment, marked as answer]
@Maxet24 Remove relative imports to `public` and instead request public assets via the base URL `/` which utilizes [public static file serving](https://nextjs.org/docs/basic-features/static-file-serving).

Working repo: https://github.com/mattcarlotta/crazy-suit-cases-fork
Working demo: https://crazy-suit-cases-fork.vercel.app/

Changes made:
[pages/_document.js](https://github.com/mattcarlotta/crazy-suit-cases-fork/blob/master/pages/_document.js#L1-L44) (optional -> preload fonts to reduce FOUC)
[pages/index.js](https://github.com/mattcarlotta/crazy-suit-cases-fork/blob/master/pages/index.js#L56-L60) (invalid nesting -> p cannot be a descendant of p)
[styles/global.css](https://github.com/mattcarlotta/crazy-suit-cases-fork/blob/master/styles/global.css#L5-L33) (request assets from `public` by utilizing the base URL `/`)

On that note, if you're trying to utilize assets across **different** domains, then this can be considered anti-pattern. Instead, you'll want to use a [CDN](https://www.cloudflare.com/learning/cdn/what-is-a-cdn/). There are ways around it, like [adding CORS to a Next project](https://vercel.com/support/articles/how-to-enable-cors#enabling-cors-in-a-next.js-app), but ideally this should be avoided.

**Reply by Maxet24** [reply]
Thanks a lot! You even created a separate repository for me. This is the best tech support ever!

**Comment by aashnamehta810** [comment]
I have tried accessing it from public by utilizing the base URL /. I am still having CORS error

**Comment by ghuachins** [comment]
Hi guys, I am having the same scenario, in which my fonts are in our design system, consume as a node module.

Error:
Access to font at 'webpack:///mini-css-extract-plugin/<app_name>/_next/static/media/Lato-Regular.eced59bb.ttf' from origin 'http://localhost:3000' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, chrome, chrome-extension, chrome-untrusted, https.

This webpack configuration in nextjs.config,js resolved the error for me:
```
webpack: (
    config,
    { isServer }
  ) => {
    config.module.generator.asset.publicPath = "/_next/";

    return config
  },
```

**Comment by DplAzim** [comment]
![image](https://github.com/user-attachments/assets/759b1032-b6ff-4ac4-9ea7-0dcd2a2a9dfa)
![image](https://github.com/user-attachments/assets/905d6c5e-21be-4cc6-acd1-223a911fdb91)


I am facing the same issue

uploaded the build file to the S3 bucket all the CSS, JS and images worked fine but the font is getting this error

import { Nunito } from "next/font/google";

//font config
const nunito_init = Nunito({
  subsets: ["latin"],
  variable: "--font-nunito",
  weight: ["200", "300", "400", "500", "600", "700", "800", "900", "1000"],
  display: "swap",
});

export const nunito = nunito_init.variable;
